### PR TITLE
chore: upgrade @eslint/eslintrc@2.0.0

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

Upgrades to yet-to-be-released `@eslint/eslintrc@2.0.0`.

#### What changes did you make? (Give an overview)

Updated the dependency in `package.json`. Also updated `eslint.config.js` to pass in now-required eslint recommended config object to `FlatCompat`, otherwise linting would fail.

#### Is there anything you'd like reviewers to focus on?

Must be updated before merging as it's currently using `@eslint/eslintrc` from GitHub.

### CODE REVIEW

File: package.json
Comment (mdjermanovic) [inline package.json:63]:
```suggestion
    "@eslint/eslintrc": "^2.0.0",
```

Before merging.

**Review by btmills — APPROVED** [review]
LGTM pending CI. Thanks for queuing this up for me!

**Pull request merged**
